Core/src/org/sleuthkit/autopsy/modules/plaso/PlasoIngestModule.java

```diff
@@ -0,0 +1,410 @@
+    private static final String PSORT_EXECUTABLE = "psort.exe";
+    private static final String VSS_OPTIONS = "--vss-stores";
+    private static final String PARTITIONS = "--partitions";
+    private static final String HASHER_FILE_SIZE_LIMIT = "--hasher_file_size_limit";
+    private static final String ONE = "1";
+    private static final String HASHERS = "--hashers";
+    private static final String NONE = "none";
+    private static final String ALL = "all";
+    private IngestJobContext context;
+
+    PlasoIngestModule() {
+    }
+
+    @Override
+    public void startUp(IngestJobContext context) throws IngestModuleException {
+        this.context = context;
+
+    }
+
+    @NbBundle.Messages({
+        "PlasoIngestModule_startUp_message=Starting Plaso Run.",
+        "PlasoIngestModule_error_running=Error running Plaso."
+    })
+
+    @Override
+    public ProcessResult process(Content dataSource, DataSourceIngestModuleProgress statusHelper) {
+        String imgName = dataSource.getName();
+        File log2TimeLineExecutable;
+        File psortExecutable;
+        Case currentCase;
+        String moduleOutputPath;
+        String tempOutputPath;
+        
+        Image img = (Image) dataSource;
+
+        currentCase = Case.getCurrentCase();
+
+        log2TimeLineExecutable = locateExecutable(LOG2TIMELINE_EXECUTABLE);
+        if (log2TimeLineExecutable == null) {
+            //TODO   Throw exeption
+        }
+        psortExecutable = locateExecutable(PSORT_EXECUTABLE);
+        if (psortExecutable == null) {
+            //TODO   Throw exeption
+        }
+        
+        String[] imgFile = img.getPaths();
+        
+        /*
+        * Make an module and temp output folder unique to this data source.
+        */
+        Long dataSourceId = dataSource.getId();
+        moduleOutputPath = Paths.get(currentCase.getModuleDirectory(), PLASO).toString();
+        File directory = new File(String.valueOf(moduleOutputPath));
+        if (!directory.exists()) {
+            directory.mkdirs();   
+        }
+        tempOutputPath = Paths.get(currentCase.getTempDirectory(), PLASO, imgName).toString();
+        directory = new File(String.valueOf(tempOutputPath));
+        if (!directory.exists()) {
+            directory.mkdirs();
+        }
+
+        ProcessBuilder log2TimeLineCommand = buildLog2TimeLineCommand(log2TimeLineExecutable, tempOutputPath, imgFile[0]);
+        ProcessBuilder psortCommand = buildPsortCommand(psortExecutable, tempOutputPath);
+        //logger.log(Level.INFO, "Hash value stored in {0}: {1}", new Object[]{imgName, storedHash}); //NON-NLS
+
+        logger.log(Level.INFO, Bundle.PlasoIngestModule_startUp_message()); //NON-NLS
+
+        try {
+            plasoWorker processImage = new plasoWorker(log2TimeLineCommand, psortCommand, moduleOutputPath, tempOutputPath);
+            processImage.execute();
+        } catch (Exception ex) {
+          //TODO Make a real exception
```

Comment:
Please resolve the TODO, which should happen when you remove the SwingWorker. FYI for future reference, SwingWorker.execute never throws an exception. Instead, the SwingWorker framework already provides an exception firewall around SwingWorker.doInBackground. To get exceptions from SwingWorker.doInBackground, you need to call get() in done().